src/cryptsetup/cryptsetup-generator.c

```diff
@@ -261,7 +259,7 @@ static int print_dependencies(FILE *f, const char* device_path, const char* time
                         return log_error_errno(r, "Failed to generate unit name: %m");
 
                 fprintf(f, "After=%1$s\n", unit);
-                if (canfail) {
+                if (canfail && timeout_value) {
```

Comment:
The else branch below also needs modification, i.e. the whole if statement should look like the following:

```c
if (!canfail)
        fprintf("Requires=...", ...);
else if (timeout_value) {
        fprintf("Wants=...", ...);
        write_device_timeout(...);
}
```

Otherwise, if canfail is true but timeout_value is NULL, a `Requires=` dependency is unexpectedly added.